exchanges/graphcache/src/extras/simplePagination.ts

```diff
@@ -74,21 +78,20 @@ export const simplePagination = ({
         continue;
       }
 
-      if (!prevOffset || currentOffset > prevOffset) {
-        for (let j = 0; j < links.length; j++) {
-          const link = links[j];
-          if (visited.has(link)) continue;
-          result.push(link);
-          visited.add(link);
-        }
+      const tempResult: NullArray<string> = [];
+
+      for (let j = 0; j < links.length; j++) {
+        const link = links[j];
+        if (visited.has(link)) continue;
+        tempResult.push(link);
+        visited.add(link);
+      }
+
+      const isAhead = !prevOffset || currentOffset > prevOffset;
+
+      if (isAhead === (mergeMode === 'inwards')) {
```

Comment:
Let's simplify this to:

```suggestion
      if ((!prevOffset || currentOffset > prevOffset) && mergeMode === 'inwards') {
```

I think that will make it a bit easier to ready, we can remove line 90 with this addition